packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -909,11 +909,6 @@ function collectParams(
           params.push(value);
         } else if (!sourceProp.model) {
           params.push(value);
-        } else {
-          // eslint-disable-next-line no-console
-          console.log(
-            `We are not counting "${sourceProp.name}" as part of a method parameter because it's been added by Azure.Core templates`,
-          );
```

Comment:
do we need to do something or report something when we ignore it?

Reply:
i removed the core model filtering logic. i believe it will cause method param mapping issues. also, we should not specialize the logic for azure core model. cc: @iscai-msft

Reply:
we did this because we didn't want users to have to deal with the injected values from core templates. is there a way to check if a property was added by a library template instead of the user?

Reply:
the filtering will cause `SdkServiceMethod` level missing params. thus i removed such logic. @catalinaperalta do you have real case here?

Reply:
@catalinaperalta i merged this pr and will release a new tcgc version soon. if any unexpected behavior happens, feel free to ping me directly.